### ISSUE TITLE
large fiberassign vs. platemaker XY mismatches

### ISSUE THREAD

**Issue by sbailey** [desihub/fiberassign#330]
In some cases there are large mismatches between the fiberassign predicted X,Y and the platemaker requested X,Y.  e.g. from @deisenstein looking at night 20210418 expid 00085353.
![image](https://user-images.githubusercontent.com/218471/115653559-375cbd80-a2e4-11eb-90c6-a22b613ba3b0.png)
From raw data headers this had ADC1PHI = 220.079999 and ADC2PHI = 224.719997 and HEXPOS  = '1154.8,-434.1,318.7,-4.5,24.0,89.7'.  The last number is the rotation about z in arcsec vs. the fiberassign header FIELDROT=-0.144462389554905 deg = -520 arcsec (which is large, but I think the max is +/- 599 arcsec).

Or a completely different example 20210412/00084534 identified by @djschlegel with a scale factor, maxing out at ~100 um discrepancy:
![image](https://user-images.githubusercontent.com/218471/115654471-0da49600-a2e6-11eb-9718-c5a5f2460c86.png)
This had ADC1PHI = 171.439997, ADC2PHI = 213.920609, and HEXPOS  = '1096.8,-719.2,61.3,13.5,22.7,-59.1' vs. fiberassign FIELDROT=-0.30833763410123 deg = -1110 arcsec (!)

UPDATE 2021-04-25: I realized I was mis-interpreting FIELDROT vs. hexapod rotation, which are different things.  FIELDROT is the rotation of CS5 +y with respect to +dec on the sky.  The fiberassign requested value of FIELDROT is what the field rotation would be on the date FA_PLAN if the hexapod has zero rotation.  i.e. FIELDROT is not the hexapod rotation, and one only expects the hexapod rotation to be zero for that field if observed on FA_PLAN (which is different from the date on which fiberassign was run).

i.e. we will have fiberassign vs. platemaker X,Y discrepancies, but HEXPOS vs. FIELDROT was a red herring.

PREVIOUS INCORRECT COMMENTS for context interpreting other comments:
> Those numbers come from the raw data and fiberassign headers, but I suspect I'm misinterpreting something because they seems to imply
>   * fiberassign is requesting FIELDROT larger than allowable
>   * platemaker isn't putting the hexapod at the requested rotation even when allowable
>   * but that discrepancy doesn't seem to be reflected in the quiver plots anyway and we don't seem to be seeing positioning failures preferentially at large radius where rotation offsets would matter.

**Comment by tskisner** [comment]
Just to review a few things.  The FIELDROT value is computed using a function from desimodel:
https://github.com/desihub/fiberassign/blob/dced43beebe4c513bde055900ff06a696f4c73dc/py/fiberassign/tiles.py#L85

The "observation time" is either overriden at run time for all tiles, or a OBSDATE column in the footprint file is used if it exists, or else a date is chose in the middle of the survey:
https://github.com/desihub/fiberassign/blob/dced43beebe4c513bde055900ff06a696f4c73dc/py/fiberassign/tiles.py#L60-L74

This angle in degrees is used in the radec2xy calculations and is written out to the fiberassign files.  For the exposures in these images, how was the observation time set?  Did the footprint file have an `OBSDATE` column?  Was a single observation time specified at run time?  Was nothing specified so that the default was used?  If the number recorded in the output is wrong, then we should print out the input / output values from the call to `field_rotation_angle()` and see if they make sense.

**Comment by sbailey** [comment]
I updated the original post — I had mixed up field rotation (rotation between CS5 +y and +dec on the sky) with hexapod rotation.  i.e. they are related but they are different things and shouldn't numerically match.  None-the-less, we still have x,y prediction mis-matches due to
* airmass "field squashing" (#331)
* ADC settings assumptions (AFAIK fiberassign doesn't try to guess what the ADC settings should be or make any higher order adjustments for that)
* scale factors (I think this is still a free parameter in the desimeter fits that isn't understood why it isn't 1.0000 and hasn't been back-propagated to desimodel/fiberassign platescale predictions)
* possibly other reasons
